src/ir_Argo.cpp

```diff
@@ -0,0 +1,282 @@
+}
+
+void IRArgoESP::checksum() {
+  uint8_t sum = 2; // Corresponds to byte 11 being constant 0b01
+  uint8_t i;
+
+  // Only add up bytes to 9. byte 10 is 0b01 constant anyway.
+  // Assume that argo array is MSB first (left)
+  for (i = 0; i < 10; i++)
+    sum += argo[i];
+
+  sum = sum % 256;  // modulo 256
+  // Append sum to end of array
+  // Set const part of checksum bit 10
+  argo[10] = 0b00000010;
+  argo[10] += sum << 2;  // Shift up 2 bits and append to byte 10
+  argo[11] = sum >> 6;  // Shift down 6 bits and add in two LSBs of bit 11
+}
+
+void IRArgoESP::stateReset() {
+  for (uint8_t i = 0; i < ARGO_COMMAND_LENGTH; i++)
+    argo[i] = 0x0;
+
+  // Argo Message. Store MSB left.
+  // Default message:
+  argo[0] = 0b10101100;  // LSB first (as sent) 0b00110101; //const preamble
+  argo[1] = 0b11110101;  // LSB first: 0b10101111; //const preamble
+  // Keep payload 2-9 at zero
+  argo[10] = 0b00000010;  // Const 01, checksum 6bit
+  argo[11] = 0b00000000;  // Checksum 2bit
+
+  this->off();
+  this->setTemp(20);
+  this->setRoomTemp(25);
+  this->setCoolMode(ARGO_COOL_AUTO);
+  this->setFan(ARGO_FAN_AUTO);
+
+  checksum();
+}
+
+uint8_t* IRArgoESP::getRaw() {
+  checksum();  // Ensure correct settings before sending.
+  return argo;
+}
+
+void IRArgoESP::on() {
+  // state = ON;
+  ac_state=1;
+  // Bit 5 of byte 9 is on/off
+  // in MSB first
+  argo[9] = argo[9] | 0b00100000;  // Set ON/OFF bit to 1
+
+  checksum();
+}
+
+void IRArgoESP::off() {
+  // state = OFF;
+  ac_state=0;
+  //in MSB first
+  //bit 5 of byte 9 to off
+  argo[9] = argo[9] & 0b11011111;  // Set on/off bit to 0
+  checksum();
+}
+
+void IRArgoESP::setPower(bool state) {
+  if (state)
+    on();
+  else
+    off();
+}
+
+uint8_t IRArgoESP::getPower() {
+  return ac_state;
+}
+
+void IRArgoESP::setMax(bool state) {
+  max_mode = state;
+  if (max_mode)
+    argo[9] |= 0b00001000;
+  else
+    argo[9] &= 0b11110111;
+  checksum();
+}
+
+bool IRArgoESP::getMax() {
+  return max_mode;
+}
+
+// Set the temp in deg C
+// Sending 0 equals +4
+void IRArgoESP::setTemp(uint8_t temp) {
+  if (temp < ARGO_MIN_TEMP)
+    temp = ARGO_MIN_TEMP;
+  else if (temp > ARGO_MAX_TEMP)
+    temp = ARGO_MAX_TEMP;
+
+  // Store in attributes
+  set_temp = temp;
+  // offset 4 degrees. "If I want 12 degrees, I need to send 8"
+  temp -= 4;
+  // Settemp = Bit 6,7 of byte 2, and bit 0-2 of byte 3
+  // mask out bits
+  // argo[13] & 0x00000100;  // mask out ON/OFF Bit
+  argo[2] &= 0b00111111;
+  argo[3] &= 0b11111000;
+
+  argo[2] += temp << 6;  // append to bit 6,7
+  argo[3] += temp >> 2;  // remove lowest to bits and append in 0-2
+  checksum();
+}
+
+uint8_t IRArgoESP::getTemp() {
+  return set_temp;
+}
+
+// Set the speed of the fan
+void IRArgoESP::setFan(uint8_t fan) {
+  // Set the fan speed bits, leave low 4 bits alone
+  fan_mode = fan;
+  // Mask out bits
+  argo[3] &= 0b11100111;
+  // Set fan mode at bit positions
+  argo[3] += fan << 3;
+  checksum();
```

Comment:
If you add this to where the other comments say, you can remove this instance of `checksum()`.